pkg/storage/etcd/etcd_helper.go

```diff
@@ -289,14 +289,20 @@ func (h *etcdHelper) decodeNodeList(nodes []*etcd.Node, slicePtr interface{}) er
 	for _, node := range nodes {
 		if node.Dir {
 			trace.Step("Decoding dir " + node.Key + " START")
-			if err := h.decodeNodeList(node.Nodes, slicePtr); err != nil {
+			if err := h.decodeNodeList(node.Nodes, filter, slicePtr); err != nil {
 				return err
 			}
 			trace.Step("Decoding dir " + node.Key + " END")
 			continue
 		}
 		if obj, found := h.getFromCache(node.ModifiedIndex); found {
-			v.Set(reflect.Append(v, reflect.ValueOf(obj).Elem()))
+			matches, err := filter(obj)
+			if err != nil {
```

Comment:
I'm fine with changing it.
Basically, I though that there is some deep reason behind why Match() function in Matcher returns false. But it seems there isn't.
I will change that today.